src/app/store/reducers/product.reducers.ts

```diff
@@ -0,0 +1,41 @@
+import {initialProductState, ProductState} from "../state/product.state";
+import {ProductActions, ProductActionsEnum} from "../actions/product.actions";
+
+export const productReducers = (
+  state = initialProductState,
+  action: ProductActions
+): ProductState =>{
+  switch (action.type) {
+    case ProductActionsEnum.GetAllProductsSuccess:{
+      return {
+        ...state,
+        products:action.payload
```

Comment:
In the future you might want to consider adding a loading flag here such that you can control how your pages are loading.